java-frontend/src/main/java/org/sonar/java/ast/JavaAstScanner.java

```diff
@@ -73,7 +74,13 @@ public void scan(Iterable<? extends InputFile> inputFiles) {
           cancelled = true;
           break;
         }
-        executionTimeReport.start(inputFile.toString());
+        int lengthInBytes = -1;
+        try {
+          lengthInBytes = inputFile.contents().length();
```

Comment:
Because some implementations of InputFile load the file content at each `contents()` call, it would be better to have `executionTimeReport.start(inputFile)` without content length computation and only compute the length in the `if (currentAnalysisTime >= minRecordedOrderedExecutionTime) {` block.